### PR TITLE
Move typescript interfaces to PHC

### PR DESCRIPTION
**Pull request by tonidero**
This PR moves most of our typescript interfaces to PHC and the new package `@revenuecat/purchases-typescript-internal`. This doesn't include the `Purchases` class itself since it's much more difficult to share that code between hybrids.

No APIs should have changed from this change, it's just moving interfaces to a different package.

### CODE REVIEW

**Review by NachoSoto — COMMENTED** [review]
😍 ship it

**Comment by aboedo** [comment]
gotta love that ratio
image

File: src/purchases.ts [hunk 1]
```diff
@@ -14,24 +18,38 @@ import {
...
+  BILLING_FEATURE,
+  REFUND_REQUEST_STATUS,
+  LOG_LEVEL,
+  PurchasesConfiguration,
+  CustomerInfoUpdateListener,
+  ShouldPurchasePromoProductListener,
+  MakePurchaseResult,
+  LogHandler,
+  LogInResult
+} from '@revenuecat/purchases-hybrid-typescript-interfaces';
```
Comment (aboedo) [inline src/purchases.ts:46]: ❤️

**Comment by tonidero** [comment]
This is ready for review @RevenueCat/coresdk

File: src/customerInfo.ts [hunk 2]
```diff
@@ -1,156 +1,2 @@
...
-     * Product Id associated with the transaction.
-     */
-    productIdentifier: string;
-    /**
-     * Purchase date of the transaction in ISO 8601 format.
-     */
-    purchaseDate: string;
-}
+// This file is kept to keep backwards compatibility to any possible users using this file directly
+export * from '@revenuecat/purchases-typescript-internal/dist/customerInfo';
```
Comment (NachoSoto) [inline src/customerInfo.ts:2]: <3

**Review by NachoSoto — APPROVED** [review]
So. Much. Deleted. Code.

**Pull request merged**